### DISCUSSION TITLE
Pagination?

### DISCUSSION THREAD

**Discussion by ffpetrovic** [simolus3/drift#1228] [Q&A]
I see that that there are potential [issues](https://github.com/simolus3/moor/issues/254) while watching an increasingly large query. What I'm looking for is pagination support (in the UI implementation it's actually a sort of infinite scroll), and I guess I'd go about it by adding a DAO method in the Database class that accepts a parameter for the `LIMIT` part of the query. And so when the user scrolls to get more entries, this `LIMIT` parameter would increase. What if in the end I end up watching a query whose LIMIT is 10000, 50000 or 100000? Are there any optimization steps I can take?

**Comment by simolus3** [comment]
You could add an offset to the limit so that you'll always watch a fixed-length chunk at a time.

```dart
const int pageSize = 50;

Selectable<...> yourQuery({int page = 1}) {
  return select(yourTable)..limit(pageSize, offset: (page - 1) * pageSize);
}
```
